shenyu-admin/src/main/resources/mappers/api-sqlmap.xml

```diff
@@ -257,4 +287,106 @@
       date_updated = #{dateUpdated,jdbcType=TIMESTAMP}
     where id = #{id,jdbcType=VARCHAR}
   </update>
+
+  <select id="existed" resultType="java.lang.Boolean">
+    select true
+    from api
+    where id = #{id} limit 1
+  </select>
+
+  <select id="selectByQuery" parameterType="org.apache.shenyu.admin.model.query.ApiQuery"
+          resultMap="BaseVOResultMap">
+      select
+        api.id,
+        api.context_path,
+        api.api_path,
+        api.http_method,
+        api.consume,
+        api.produce,
+        api.version,
+        api.rpc_type,
+        api.`state`,
+        api.ext,
+        api.api_owner,
+        api.api_desc,
+        api.api_source,
+        api.document,
+        api.document_md5,
+        api.date_created,
+        api.date_updated,
+        tag.id as tag_id,
+        tag.date_created as tag_date_created,
+        tag.date_updated as tag_date_updated,
+        tag.name as tag_name,
+        tag.tag_desc as tag_desc,
+        tag.parent_tag_id as tag_parent_tag_id,
+        tag.ext as tag_ext
+      from api
+      left join tag_relation on api.id = tag_relation.api_id
+      left join tag on tag_relation.tag_id = tag.id
+      <where>
+          <if test="apiPath != null and apiPath != ''">
+              <bind name="nameLike" value="('%' + apiPath + '%')"/>
+              AND api_path LIKE #{nameLike, jdbcType=VARCHAR}
+          </if>
+          <if test="state != null">
+              AND state = #{state, jdbcType=TINYINT}
+          </if>
+          <if test="tagId != null and tagId != ''">
+            and tag.id = #{tagId,jdbcType=VARCHAR}
+          </if>
+      </where>
+      order by api.date_created desc
+  </select>
+
+  <select id="selectByIds" parameterType="java.util.List" resultMap="BaseResultMap">
+    SELECT
+    <include refid="Base_Column_List"/>
+    FROM api
+    WHERE id IN
+    <foreach item="id" collection="list" open="(" separator="," close=")">
+      #{id, jdbcType=VARCHAR}
+    </foreach>
+  </select>
+
+  <delete id="deleteByIds" parameterType="java.util.List">
+    DELETE FROM api
+    WHERE id IN
+    <foreach item="id" collection="list" open="(" separator="," close=")">
+      #{id, jdbcType=VARCHAR}
+    </foreach>
+  </delete>
+
+  <select id="selectByPrimaryKeyGetVO" resultMap="BaseVOResultMap">
+    select
+      api.id,
+      api.context_path,
+      api.api_path,
+      api.http_method,
+      api.consume,
+      api.produce,
+      api.version,
+      api.rpc_type,
+      api.`state`,
+      api.ext,
+      api.api_owner,
+      api.api_desc,
+      api.api_source,
+      api.document,
+      api.document_md5,
+      api.date_created,
+      api.date_updated,
+      tag.id as tag_id,
+      tag.date_created as tag_date_created,
+      tag.date_updated as tag_date_updated,
+      tag.name as tag_name,
+      tag.tag_desc as tag_desc,
+      tag.parent_tag_id as tag_parent_tag_id,
+      tag.ext as tag_ext
```

Comment:
`api-sqlmap` only handles `ApiDO` and should not query from `tag_relation` or `tag`, it is recommended to assemble complex `VOs` on the `Service` layer.

Reply:
fixed it